core/src/plugins/filed/python/postgresql/bareos-fd-postgresql.py

```diff
@@ -0,0 +1,1396 @@
+    if int(major) < 1 or (int(major) == 1 and int(minor) < 16):
+        raise ValueError(
+            (
+                f"pg8000 module version ({pg8000.__version__}) is lower"
+                f" than required version 1.16\n"
+            )
+        )
+except ValueError as err_version:
+    bareosfd.JobMessage(bareosfd.M_FATAL, f"FATAL ERROR: {err_version}\n")
+except ImportError as err_import:
+    bareosfd.JobMessage(
+        bareosfd.M_FATAL, f"could not import Python module: pg8000 {err_import}\n"
+    )
+except Exception as err_unknown:
+    bareosfd.JobMessage(bareosfd.M_FATAL, f"Unknown error {err_unknown}\n")
+
+
+def parse_row(row):
+    """
+    This function exists to fix bug in pg8000 >= 1.26 < 1.30.0
+    Returned results of pg_backup_stop are string representation
+    of tuple
+    """
+    remove_parens = row[1:-1]
+    [lsn, backup_label, tablespace] = remove_parens.split(",")
+    # lsn does not have quotes
+    # remove quotes
+    backup_label = backup_label[1:-1]
+    tablespace = tablespace[1:-1]
+    return [lsn, backup_label, tablespace]
+
+
+@BareosPlugin
+class BareosFdPluginPostgreSQL(BareosFdPluginBaseclass):  # noqa
+    """
+    Bareos-FD-Plugin-Class for PostgreSQL online (Hot) backup of cluster
+    files and database transaction logs (WAL) archiving to allow incremental
+    backups and point-in-time (pitr) recovery.
+    If the cluster use tablespace, the backup will also backup and restore those
+    (symlinks in pg_tblspc, and real external location data)
+    The plugin job will fail if previous job was not done on same PG major version
+    """
+
+    def __init__(self, plugindef):
+        bareosfd.DebugMessage(
+            100, f"Constructor called in module {__name__} with plugindef={plugindef}\n"
+        )
+
+        bareosfd.DebugMessage(
+            100,
+            (
+                f"Python Version: {version_info.major}.{version_info.minor}"
+                f".{version_info.micro}\n"
+            ),
+        )
+
+        # Last argument of super constructor is a list of mandatory arguments
+        super().__init__(plugindef, ["postgresql_data_dir", "wal_archive_dir"])
+
+        # The contents of the directories pg_dynshmem/, pg_notify/, pg_serial/, pg_snapshots/,
+        # pg_stat_tmp/, and pg_subtrans/ can be omitted from the backup as they will be initialized
+        # on postmaster startup. (but not the directories themselves)
+        # They are mandatory to make pg cluster start successful.
+        # By default we will exclude them so os.walk will not go into it,
+        # but we re-add the dir afterwards in case of Full backup.
+        self.mandatory_subdirs = [
+            "pg_dynshmem",
+            "pg_notify",
+            "pg_serial",
+            "pg_snapshots",
+            "pg_stat_tmp",
+            "pg_subtrans",
+            "pg_wal",
+        ]
+        # log is not exclude by default but can be a good candidate.
```

Comment:
Picked.

Not as simple it look first. Some people won't want to have the log in the backup and the restore, while other would like to have them. So I would prefer to let it like this, as it can serve as documentation for people reading the code instead the manual ;-)